pkg/devfile/devfile.go

```diff
@@ -432,6 +480,72 @@ func GetResourceFromDevfile(log logr.Logger, devfileData data.DevfileData, deplo
 	return appendedResources, err
 }
 
+// GetIngressFromEndpoint gets an ingress resource from the devfile endpoint information
+func GetIngressFromEndpoint(name, serviceName, port, path string, secure bool, annotations map[string]string, namespace, ingressDomain string) (networkingv1.Ingress, error) {
+
+	if path == "" {
+		path = "/"
+	}
+
+	implementationSpecific := networkingv1.PathTypeImplementationSpecific
+
+	portNumber, err := strconv.Atoi(port)
+	if err != nil {
+		return networkingv1.Ingress{}, nil
+	}
+
+	hostname, err := getIngressHostName(serviceName, namespace, ingressDomain)
+	if err != nil {
+		return networkingv1.Ingress{}, nil
+	}
+
+	ingress := networkingv1.Ingress{
+		TypeMeta:   generator.GetTypeMeta("Ingress", "networking.k8s.io/v1"),
+		ObjectMeta: generator.GetObjectMeta(name, "", nil, nil),
+		Spec: networkingv1.IngressSpec{
+			Rules: []networkingv1.IngressRule{
+				{
+					Host: hostname,
+					IngressRuleValue: networkingv1.IngressRuleValue{
+						HTTP: &networkingv1.HTTPIngressRuleValue{
+							Paths: []networkingv1.HTTPIngressPath{
+								{
+									Path:     path,
+									PathType: &implementationSpecific,
+									Backend: networkingv1.IngressBackend{
+										Service: &networkingv1.IngressServiceBackend{
+											Name: serviceName,
+											Port: networkingv1.ServiceBackendPort{
+												Number: int32(portNumber),
```

Comment:
Got me thinking... what if the Component spec targret port is different from the kubernetes endpoint port that we based the ingress/route off of? I think `spec.targetPort` should have priority over the Kubernetes endpoint port if it's set here.

Reply:
yes we do it here https://github.com/redhat-appstudio/application-service/pull/315/files#diff-3c96648f13e4abf5dd637ead679b75d28c42c97a4f5c4509ac1a1f53ecf5307bR458